crate/server/src/core/operations/set_attribute.rs

```diff
@@ -0,0 +1,130 @@
+            attributes.activation_date = Some(activation_date);
+        }
+        Attribute::CryptographicAlgorithm(cryptographic_algorithm) => {
+            trace!(
+                "Set Attribute: Cryptographic Algorithm: {:?}",
+                cryptographic_algorithm
+            );
+            attributes.cryptographic_algorithm = Some(cryptographic_algorithm);
+        }
+        Attribute::CryptographicLength(length) => {
+            trace!("Set Attribute: Cryptographic Length: {:?}", length);
+            attributes.cryptographic_length = Some(length);
+        }
+        Attribute::CryptographicParameters(parameters) => {
+            trace!("Set Attribute: Cryptographic Parameters: {:?}", parameters);
+            attributes.cryptographic_parameters = Some(Box::new(parameters));
+        }
+        Attribute::CryptographicDomainParameters(domain_parameters) => {
+            trace!(
+                "Set Attribute: Cryptographic Domain Parameters: {:?}",
+                domain_parameters
+            );
+            attributes.cryptographic_domain_parameters = Some(domain_parameters);
+        }
+        Attribute::CryptographicUsageMask(usage_mask) => {
+            trace!("Set Attribute: Cryptographic Usage Mask: {:?}", usage_mask);
+            attributes.cryptographic_usage_mask = Some(usage_mask);
+        }
+        Attribute::Links(links) => {
+            trace!("Set Attribute: Link: {:?}", links);
+            for link in &links {
+                attributes.set_link(link.link_type, link.linked_object_identifier.clone());
+            }
+        }
+        Attribute::State(state) => {
+            trace!("Set Attribute: State: {:?}", state);
+            owm.state = state;
+        }
+        Attribute::VendorAttributes(vendor_attributes) => {
+            trace!("Set Attribute: Vendor Attributes: {:?}", vendor_attributes);
+            for vendor_attribute in vendor_attributes {
+                attributes.set_vendor_attribute(
+                    &vendor_attribute.vendor_identification,
+                    &vendor_attribute.attribute_name,
+                    vendor_attribute.attribute_value,
+                );
+            }
+        }
+    }
+
+    let tags = kms.db.retrieve_tags(&owm.id, params).await?;
+
+    match owm.object.object_type() {
+        ObjectType::PublicKey
+        | ObjectType::PrivateKey
+        | ObjectType::SplitKey
+        | ObjectType::SecretData
+        | ObjectType::PGPKey
+        | ObjectType::SymmetricKey => {
+            let object_attributes = owm.object.attributes_mut()?;
+            *object_attributes = attributes.clone();
+            debug!("Set Object Attribute: {:?}", object_attributes);
+        }
+        _ => {
+            trace!(
+                "Set Attribute: Object type {:?} does not have attributes (nor key block)",
+                owm.object.object_type()
+            );
+        }
+    }
+
+    debug!("Set Attribute: {:?}", attributes);
```

Comment:
Set attributes ? (plural) ?

Reply:
In server, the SetAttribute operation is a single attribute modification. 
In ckms, the SetAttributes operation call one or multiple times the SetAttribute